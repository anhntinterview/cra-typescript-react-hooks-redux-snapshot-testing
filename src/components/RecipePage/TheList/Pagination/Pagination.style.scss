.pagination {
  background-color: #fff;
  bottom: 0;
  display: flex;
  flex-direction: row;
  width: 125px;
  margin: 1rem auto;

  .item {
    padding: 0.5rem;
    border: 1px solid #ccc;
    border-radius: 0.5rem;
    width: 20px;
    margin-right: 5px;
    text-align: center;
    cursor: pointer;
    &:hover {
        background-color: #ccc;
        border: 1.5px solid #555;
    }
    &.current {
        background-color: #eee;
        border: 1.3px solid #333;
    }
  }
}
