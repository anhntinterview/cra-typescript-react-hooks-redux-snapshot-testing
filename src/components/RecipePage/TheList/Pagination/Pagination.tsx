import * as React from "react";
import "./Pagination.style.scss";

export interface PaginationProps {
  currentPage: number;
  totalPages: number;
  handleCurrenPage: (param: number) => void;
}

export function renderArray(totalPages: number) {
  const temp = [];
  let theIndex = 1;
  while (theIndex <= totalPages) {
    temp.push(theIndex);
    theIndex++;
  }
  return temp;
}

export function handleClick(
  handleCurrenPage: (param: number) => void,
  item: number
) {
  return () => {
    handleCurrenPage(item);
  };
}

const Pagination: React.FC<PaginationProps> = ({
  totalPages,
  handleCurrenPage,
  currentPage
}) => {
  const pagination = renderArray(totalPages);

  return (
    <ul className="pagination">
      {pagination.map((item, index) => (
        <li className={`item ${(index+1 === currentPage ? 'current' : '')}`} key={index} onClick={handleClick(handleCurrenPage, item)}>
          {item}
        </li>
      ))}
    </ul>
  );
};

export default Pagination;
