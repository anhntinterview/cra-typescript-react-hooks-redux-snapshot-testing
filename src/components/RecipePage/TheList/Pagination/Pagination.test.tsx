import * as React from "react";
import { 
  shallow, 
} from "enzyme";

import Pagination, {PaginationProps} from "./Pagination";

const props: PaginationProps ={ 
    currentPage: 1,
  totalPages: 5,
  handleCurrenPage: () => {
      return null
  }
}

describe('Pagination', () => {
  it('should match a snapshot', () => {
    expect(shallow(<Pagination {...props}/>)).toMatchSnapshot();
  })
})
