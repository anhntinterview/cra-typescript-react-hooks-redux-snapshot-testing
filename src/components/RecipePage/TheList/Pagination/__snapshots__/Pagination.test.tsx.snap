// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match a snapshot 1`] = `
<ul
  className="pagination"
/>
`;
